OpenDreamShared/Net/ClientData.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.IO;
+using System.Runtime.Serialization;
+
+namespace OpenDreamShared.Net {
+    [Serializable]
```

Comment:
`IconAppearance` has  `WriteToPacket` and `ReadFromPacket` methods that use a `PacketStream` to serialize/deserialize. Could you do something similar here instead of `[Serializable]`?